tests/integration_test/core_mqtt_system_test.c

```diff
@@ -1562,6 +1569,17 @@ void Restore_Session_Duplicate_Incoming_Publish_Qos1()
     /* Make sure that a record was created for the incoming PUBLISH packet. */
     TEST_ASSERT_NOT_EQUAL( MQTT_PACKET_ID_INVALID, context.incomingPublishRecords[ 0 ].packetId );
 
+    if( testingAgainstAWS )
+    {
+        /* AWS IoT Core takes at least 30 seconds before resending a PUBLISH. */
+        vTaskDelay( pdMS_TO_TICKS( AWS_IOT_CORE_REPUBLISH_INTERVAL_MS ) );
+
+        /* When the resent PUBLISH is received, the DUP flag will not be set to 1.
+         * Therefore, the packet ID of the original PUBLISH must be cleared from
+         * the incoming publish records array. */
+        context.incomingPublishRecords[ 0 ].packetId = MQTT_PACKET_ID_INVALID;
```

Comment:
Why is this required for the `MQTTContext_t`. According to the spec, even if the DUP flag is set, it does not mean that the receiving end had received the original PUBLISH message..

Reply:
IoT Core will not set the DUP flag in the resent publish. However, during the first time we receive the publish, the state array is updated. Therefore, when we receive it again and the DUP flag is not set, the library will return a `MQTTStateCollision` error.